src/util/psbt/mod.rs

```diff
@@ -16,6 +21,138 @@ pub mod serialize;
+    /// Extract a Transaction from a finalized PartiallySignedTransaction
+    pub fn extract_tx(self) -> Transaction {
+        let mut tx: Transaction = self.global.unsigned_tx;
+
+        for (vin, psbtin) in tx.input.iter_mut().zip(self.inputs.into_iter()) {
+            vin.script_sig = psbtin.final_script_sig.unwrap();
+            vin.witness = psbtin.final_script_witness.unwrap();
+        }
+
+        tx
+    }
+
+    /// Attempt to merge with another `PartiallySignedTransaction`.
+    pub fn merge(&mut self, other: Self) -> Result<(), util::Error> {
+        self.global.merge(other.global)?;
+
+        for (self_input, other_input) in self.inputs.iter_mut().zip(other.inputs.into_iter()) {
+            self_input.merge(other_input)?;
+        }
+
+        for (self_output, other_output) in self.outputs.iter_mut().zip(other.outputs.into_iter()) {
+            self_output.merge(other_output)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl<S: SimpleEncoder<Error = util::Error>> ConsensusEncodable<S> for PartiallySignedTransaction {
+    fn consensus_encode(&self, s: &mut S) -> Result<(), S::Error> {
+        [b'p', b's', b'b', b't'].consensus_encode(s)?;
+
+        0xff_u8.consensus_encode(s)?;
+
+        self.global.consensus_encode(s)?;
+
+        for i in &self.inputs {
+            i.consensus_encode(s)?;
+        }
+
+        for i in &self.outputs {
+            i.consensus_encode(s)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl<D: SimpleDecoder<Error = util::Error>> ConsensusDecodable<D> for PartiallySignedTransaction {
+    fn consensus_decode(d: &mut D) -> Result<Self, D::Error> {
+        let magic: [u8; 4] = ConsensusDecodable::consensus_decode(d)?;
+
+        if [b'p', b's', b'b', b't'] != magic {
+            return Err(D::Error::from(Error::InvalidMagic));
+        }
+
+        if 0xff_u8 != ConsensusDecodable::consensus_decode(d)? {
+            return Err(D::Error::from(Error::InvalidSeparator));
+        }
+
+        let global: PsbtGlobal = ConsensusDecodable::consensus_decode(d)?;
+
+        let inputs: Vec<PsbtInput> = {
+            let inputs_len: usize = (&global.unsigned_tx.input).len();
+
+            let mut inputs: Vec<PsbtInput> = Vec::with_capacity(inputs_len);
+
+            for _ in 0..inputs_len {
+                inputs.push(ConsensusDecodable::consensus_decode(d)?);
+            }
+
+            inputs
+        };
+
+        let outputs: Vec<PsbtOutput> = {
+            let outputs_len: usize = (&global.unsigned_tx.output).len();
+
+            let mut outputs: Vec<PsbtOutput> = Vec::with_capacity(outputs_len);
+
+            for _ in 0..outputs_len {
+                outputs.push(ConsensusDecodable::consensus_decode(d)?);
+            }
+
+            outputs
+        };
+
+        Ok(PartiallySignedTransaction {
+            global: global,
+            inputs: inputs,
+            outputs: outputs,
+        })
+    }
+}
+
+impl From<Transaction> for Result<PartiallySignedTransaction, util::Error> {
```

Comment:
Is it a standard Rust idiom to impl `From` on a `Result`? I find it very confusing, `From` is supposed to be for things that are conceptually trivial conversions or reinterpretations. But a `Transaction` is nothing at all like a `Result<T, E>`.

I would prefer that `PartiallySignedTransaction` had a `from_transaction` method on it.

Reply:
I find it weird too, seems confunsing

Reply:
Agreed, I was trying to emulate the new `TryFrom` trait that isn't in 1.14.0, but will just use the dedicated method.